src/NuGet.Core/NuGet.Protocol/PackagesFolder/LocalPackageFileCache.cs

```diff
@@ -95,6 +95,20 @@ public virtual bool Sha512Exists(string sha512Path)
             return exists;
         }
 
+        internal void UpdateLastAccessTime(string nupkgMetadataPath)
+        {
+            try
+            {
+                File.SetLastAccessTimeUtc(nupkgMetadataPath, DateTime.UtcNow);
+            }
+            catch// (Exception ex)
```

Comment:
Did you mean to leave this here? Seems to match other related code.

Reply:
> - Not sure if we want to pass the logger through to NuGetv3LocalRepository.

I was not sure whether we wanted to pass the logger all the way to here, that's why I commented out the code. Would it be OK or do you have a better idea?

Reply:
I think it should be passed through. Probably as a constructor parameter, rather than a method parameter. But then that's a breaking public API change, so this method will have to handle the scenario that the logger is null 😞.  Sometimes I really hate being both a tool and an API.

Reply:
Ok, thank you for the hint. Still looking into this one, as sometimes `LocalPackageFileCache` is created before the logger instance is available.

Reply:
```
/*NuGet.Commands.*/"Strings.Error_CouldNotUpdateMetadataLastAccessTime"
```
I'm guessing we don't want to add a reference to NuGet.Commands just to get access to a string resource. Looking into alternatives...